File: Movie/Movie.Web/Modules/Club/Employee/EmployeeColumns.cs
using Movie.Modules.MovieDB.Person;
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace Movie.Club.Columns;

[ColumnsScript("Club.Employee")]
[BasedOnRow(typeof(EmployeeRow), CheckNames = true)]
public class EmployeeColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int EmployeeId { get; set; }
    [EditLink]
    public string FirstName { get; set; }
    public string LastName { get; set; }

    public string ClubName { get; set; }

    public DateTime BirthDate { get; set; }
    public string Address { get; set; }
    public Gender Gender { get; set; }
    public decimal Salary { get; set; }
}

File: Movie/Movie.Web/Modules/ServerTypes/Club/EmployeeColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { Gender } from "../Modules/MovieDB.Person.Gender";
import { EmployeeRow } from "./EmployeeRow";

export interface EmployeeColumns {
    EmployeeId: Column<EmployeeRow>;
    FirstName: Column<EmployeeRow>;
    LastName: Column<EmployeeRow>;
    ClubName: Column<EmployeeRow>;
    BirthDate: Column<EmployeeRow>;
    Address: Column<EmployeeRow>;
    Gender: Column<EmployeeRow>;
    Salary: Column<EmployeeRow>;
}

export class EmployeeColumns extends ColumnsBase<EmployeeRow> {
    static readonly columnsKey = 'Club.Employee';
    static readonly Fields = fieldsProxy<EmployeeColumns>();
}

queueMicrotask(() => [Gender]); // referenced types